docs/book/user-guide/tutorials/trigger-pipelines-from-external-systems.md

```diff
@@ -0,0 +1,590 @@
+            "run_id": run.id,
+            "template_id": template.id,
+            "template_name": template.name
+        }
+    except Exception as e:
+        raise HTTPException(
+            status_code=500,
+            detail=f"Failed to trigger template: {str(e)}"
+        )
+```
+
+### Deploying Your Custom API
+
+Deploy the API service to make it accessible to external systems:
+
+```bash
+# Install required dependencies
+pip install fastapi uvicorn
+
+# Set a secure API key
+export PIPELINE_API_KEY="your-secure-api-key-here"
+
+# Start the API server
+python pipeline_api.py
+```
+
+For production, consider deploying as a containerized service:
+
+```dockerfile
+FROM python:3.9-slim
+
+WORKDIR /app
+
+# Install ZenML and other dependencies
+COPY requirements.txt .
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Copy your code
+COPY . .
+
+# Set environment variables
+ENV PYTHONPATH=/app
+ENV PIPELINE_API_KEY=${PIPELINE_API_KEY}
+
+# Expose the port
+EXPOSE 8000
+
+# Run the API
+CMD ["python", "pipeline_api.py"]
+```
+
+### Triggering Your Pipeline via the Custom API
+
+You can [trigger pipelines](https://docs.zenml.io/how-to/trigger-pipelines) in two ways through the custom API:
+
+#### 1. Direct Pipeline Execution
+
+This method runs the pipeline directly with the specified parameters:
+
+```bash
+curl -X 'POST' \
+  'http://your-api-server:8000/trigger' \
+  -H 'accept: application/json' \
+  -H 'X-API-Key: your-secure-api-key-here' \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "pipeline_name": "training_pipeline",
+    "steps": {
+      "load_data": {
+        "parameters": {
+          "data_url": "s3://some-bucket/new-data.csv"
+        }
+      },
+      "train_model": {
+        "parameters": {
+          "model_type": "gradient_boosting"
+        }
+      }
+    }
+  }'
+```
+
+#### 2. Run Template Execution
+
+{% hint style="success" %}
+This is a [ZenML Pro](https://zenml.io/pro)-only feature. Please [sign up here](https://cloud.zenml.io) to get access.
+{% endhint %}
+
+This method uses a pre-configured run template, which is ideal for production scenarios:
+
+```bash
+curl -X 'POST' \
+  'http://your-api-server:8000/trigger-template' \
+  -H 'accept: application/json' \
+  -H 'X-API-Key: your-secure-api-key-here' \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "template_id": "YOUR_TEMPLATE_ID",
+    "steps": {
+      "load_data": {
+        "parameters": {
+          "data_url": "s3://some-bucket/new-data.csv"
+        }
+      },
+      "train_model": {
+        "parameters": {
+          "model_type": "gradient_boosting"
+        }
+      }
+    }
+  }'
+```
+
+The template-based approach offers several advantages:
+- Consistent pipeline configuration across runs
+- Pre-configured stack and environment settings
+- Centralized management of pipeline versions
+- Better control over production deployments
+
+## Best Practices & Troubleshooting
```

Comment:
maybe add a best practice of tagging run templates and making sure the step parameters don't  change otherwise upsteam things might break too